runtime/src/main/java/org/corfudb/protocols/wireprotocol/orchestrator/OrchestratorRequestType.java

```diff
@@ -0,0 +1,53 @@
+package org.corfudb.protocols.wireprotocol.orchestrator;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * The type of requests that can be made to the Orchestrator Service.
+ * 
+ * @author Maithem
+ */
+@AllArgsConstructor
+public enum OrchestratorRequestType {
+
+    /**
+     * Query a workflow id
+     */
+    QUERY(0),
+
+    /**
+     * Add a new node to the cluster
+     */
+    ADD_NODE(1);
+
+    @Getter
+    public int type;
+
+    /**
+     * Create a workflow instance from an orchestrator request
+     *
+     * @param request Orchestrator request
+     * @return Workflow instance
+     */
+    public AbstractWorkflow getWorkflow(OrchestratorRequest request) {
+        Request payload = request.getRequest();
+        switch (payload.getType()) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)

Reply:
Can you add the payload type into Request? request.getNewOrchrestrator wouldd eliminate the need for this switch.

Reply:
This comment is not clear. The Request type already has a type of its payload.

Reply:
If Request had a method ```getWorkflow``` it would eliminate the need for all this logic (which also is a maintenance nightmare).

Reply:
Are you suggesting to move this method to into Request, or eliminate the switch statement?

Reply:
Im suggesting eliminating the switch statement by adding a method to Request which returns the correct type.

Reply:
"adding a method to Request which returns the correct type", how do you suggest that method do the mapping, reflection ?

Reply:
Why would reflection be needed?

```java
public enum OrcherstratorRequestType {
ADD_NODE(1, AddNodeWorkflow::new)

final byte typeNum;
final Function<Workflow, Request> workflowGenerator;
}
```


Then here:
```java
final Request requested = request.payload;
return requested.getType().getWorkflow(requested);
```

Also, the multiple types named request are REALLY confusing. But please address this first.